### DISCUSSION TITLE
Releases should contain source tarball with the submodules included

### DISCUSSION THREAD

**Discussion by Cloudef** [LizardByte/community#24] [Sunshine]
Currently it's impossible to package sunshine from source tarballs as they don't contain the submodules

**Comment by ReenigneArcher** [comment]
> Currently it's impossible to package sunshine from source tarballs as they don't contain the submodules

The tarballs are created by github, not us. As far I understand there's no way to configure that.

**Reply by ReenigneArcher** [reply]
Of course you can `git clone` the repo, then `checkout` the specific tag recursing submodules.

**Reply by Cloudef** [reply]
Yes, I know that and it sucks github does not do anything about it. The suggested workaround is to include a tarball created by CI / manually which includes the submodules.

**Reply by ReenigneArcher** [reply]
> Yes, I know that and it sucks github does not do anything about it. The suggested workaround is to include a tarball created by CI / manually which includes the submodules.

Unless you provide a strong reason why you don't just want to use `git`, I don't plan on creating a tarball manually.

**Reply by Cloudef** [reply]
Some reasons:
1. It allows PGP signed releases https://wiki.debian.org/Creating%20signed%20GitHub%20releases
2. The source tarball can be mirrored and archived easily
3. Since the source tarball is frozen state of the project, the build can't break due to external factors (github going down, one of the submodules goes down, someone rewrites git history, etc...)
4. It allows offline builds, with integrity checks
5. Some distros outright won't allow packages not build from the source tarballs
